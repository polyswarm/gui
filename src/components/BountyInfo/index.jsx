// Vendor imports
import React, { Component } from 'react';
import PropTypes from 'prop-types';
// Bounty imports
import AssertionList from '../AssertionList';
import BountySummary from '../BountySummary';

class BountyInfo extends Component {

  render() {
    const { props: { bounty } } = this;

    return (
      <div className='Bounty-Info'>
        <div className='Bounty-Info-Container'>
          <BountySummary bounty={bounty} />
          <AssertionList className='Bounty-Info-Assertions'
            bounty={bounty} />
        </div>
      </div>
    );
  }
}

BountyInfo.propTypes = {
  bounty: PropTypes.object.isRequired,
};
export default BountyInfo;
