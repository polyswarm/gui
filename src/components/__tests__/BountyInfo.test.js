import React from 'react';
import {render, mount} from 'enzyme';
import {renderToJson} from 'enzyme-to-json';
import BountyInfo from '../BountyInfo';

it('renders without crashing', () => {
  const bounty = { };
  const wrapper = render(<BountyInfo bounty={bounty}/>);
  expect(renderToJson(wrapper)).toMatchSnapshot();
});