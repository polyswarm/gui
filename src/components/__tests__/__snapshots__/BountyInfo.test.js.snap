// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  class="Bounty-Info"
>
  <header
    class="Header"
  >
    <div
      class="Header-Row"
    >
      <div
        class="Header-Title"
      >
        <img
          alt="PolySwarm logo"
          class="Header-Logo"
          src="../public/img/back-arrow.svg"
        >
          
        </img>
        <h3>
          
        </h3>
        <span
          class="RequestSpinner"
        >
          
        </span>
      </div>
      <div
        class="Header-Actions"
      >
        
      </div>
    </div>
    <div
      class="Header-Address"
    >
      <p
        class="Header-Divided"
        style="font-size:7.0588235294117645vw"
      >
        author
      </p>
      <p
        class="Header-Divided"
        style="font-size:7.0588235294117645vw"
      >
        1 NCT 
      </p>
      <p
        style="font-size:7.0588235294117645vw"
      >
        1 ETH
      </p>
    </div>
  </header>
  <div
    class="Bounty-Info-Container"
  >
    <div
      class="BountySummary"
    >
      <li
        class="StatRow-Vertical"
      >
        <p
          class="StatTitle"
        >
          Posted By:
        </p>
        <p
          class="StatContent"
        >
          
        </p>
      </li>
      <li
        class="StatRow-Vertical"
      >
        <p
          class="StatTitle"
        >
          Resolved:
        </p>
        <p
          class="StatContent"
        >
          No
        </p>
      </li>
      <li
        class="StatRow-Vertical"
      >
        <p
          class="StatTitle"
        >
          Expired:
        </p>
        <p
          class="StatContent"
        >
          No
        </p>
      </li>
      <li
        class="StatRow-Vertical"
      >
        <p
          class="StatTitle"
        >
          Reward:
        </p>
        <p
          class="StatContent"
        >
          undefined Nectar (NCT)
        </p>
      </li>
      <li
        class="StatRow-Vertical"
      >
        <p
          class="StatTitle"
        >
          Assertions:
        </p>
        <p
          class="StatContent"
        >
          0
        </p>
      </li>
      <li
        class="StatRow-Vertical"
      >
        <p
          class="StatTitle"
        >
          IPFS Uri
        </p>
        <p
          class="StatContent"
        >
          
        </p>
      </li>
    </div>
    <div
      class="AssertionList"
    >
      <ul>
        
      </ul>
      <div
        class="Assertion-Placeholder"
      >
        <h3>
          No experts have posted any assertions. Waiting for results.
        </h3>
      </div>
    </div>
  </div>
</div>
`;
